### PR TITLE
Use Timber for logging

### PR DESCRIPTION
**Pull request by goober**
Replace current logging with [Timber](https://github.com/JakeWharton/timber). As suggested by @walles

### CODE REVIEW

File: app/src/main/java/com/liato/bankdroid/BankdroidApplication.java [hunk 1]
```diff
@@ -24,22 +22,16 @@
...
-                    bankStringBuilder.substring(0, bankStringBuilder.length() - 1));
+        CrashlyticsCore core = new CrashlyticsCore.Builder()
+                .disabled(BuildConfig.DEBUG)
+                .build();
+        Fabric.with(this, new Crashlytics.Builder().core(core).build());
+
+        if (BuildConfig.DEBUG) {
+            Timber.plant(new Timber.DebugTree());
         }
+        Timber.plant(new CrashlyticsTree());
```
Comment (walles) [inline app/src/main/java/com/liato/bankdroid/BankdroidApplication.java]: The logic you want is rather "Crashlytics on devices, not-Crashlytics on the emulator, DEBUG-vs-RELEASE is not a factor".

Also, this needs to be done at all entry points to the program; the services as well. Check `AndroidManifest.xml` for inspiration, basically all `onCreate()` implementations should set up Crashlytics.

Feel free to copy code from / copy the entire files from here and relicense it for Bankdroid if you want:
https://github.com/walles/batterylogger/blob/master/src/main/java/com/gmail/walles/johan/batterylogger/LoggingUtils.java
https://github.com/walles/batterylogger/blob/master/src/main/java/com/gmail/walles/johan/batterylogger/EmulatorUtils.java

Comment (goober) [inline app/src/main/java/com/liato/bankdroid/BankdroidApplication.java]: Thanks for the feedback. I will definitely look at your implementation for inspiration.
I also found some good stuff in your build.gradle file related to automation of versioning and release management. do you mind if I use some of that for bankdroid as well?

Comment (walles) [inline app/src/main/java/com/liato/bankdroid/BankdroidApplication.java]: Sure, I got them from StackOverflow anyways, sharing is caring :)

Comment (walles) [inline app/src/main/java/com/liato/bankdroid/BankdroidApplication.java]: You might consider not doing Crashlytics during unit testing either:
https://github.com/walles/exactype/blob/master/app/src/main/java/com/gmail/walles/johan/exactype/util/LoggingUtils.java#L64

Comment (goober) [inline app/src/main/java/com/liato/bankdroid/BankdroidApplication.java]: Thanks for the use case. I will verify the widget behaviour with `Application#onCreate()`

Comment (goober) [inline app/src/main/java/com/liato/bankdroid/BankdroidApplication.java]: @walles I have verified that the `Application#onCreate` is called for your suggested use case. So I will keep the initialization of the logger there.

**Comment by goober** [comment]
@walles I have moved the logging creation to a LoggingUtils as suggested. Please take a look before I'll merge it.

**Comment by walles** [comment]
Smashing, merge it :)

**Pull request merged**